### PR TITLE
Products

### PR DESCRIPTION
**Pull request by asherifhegazy**
need to fix the carousel to be responsive.

add to cart functionality will be implemented in cart branch

### CODE REVIEW

File: app/components/productDetails/productDetailsService.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+eShopApp.factory('productDetails',function ($http) {
+    return {
+        getProductDetails: function (id) {
```
Comment (dosama1integrant) [inline app/components/productDetails/productDetailsService.js:3]: the "https://localhost:44302/api" part need to be defined as configuration .. we can add app.config.js file to define all configurations on the application as constants and then inject it here to use the defined configuration values from it

File: app/components/products/productsService.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+eShopApp.factory('products', function ($http) {
+    return {
+        getProducts: function () {
```
Comment (dosama1integrant) [inline app/components/products/productsService.js:3]: the "https://localhost:44302/api" part need to be defined as configuration .. we can add app.config.js file to define all configurations on the application as constants and then inject it here to use the defined configuration values from it

File: app/shared/filters/truncateString.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+eShopApp.filter('truncateStr',function () {
```
Comment (dosama1integrant) [inline app/shared/filters/truncateString.js:1]: Good Handling :)

**Review by dosama1integrant — CHANGES_REQUESTED** [review]

File: app/components/products/productsService.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+            return $http({method: 'GET', url: 'https://localhost:44302/api/Products'});
```
Comment (dosama1integrant) [inline app/components/products/productsService.js]: the api reference still constant

File: app/components/productDetails/productDetailsService.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+            return $http({method: 'GET', url: 'https://localhost:44302/api/Products/' + id});
```
Comment (dosama1integrant) [inline app/components/productDetails/productDetailsService.js]: The api reference still constant

**Review by dosama1integrant — CHANGES_REQUESTED** [review]

File: app/components/products/productsService.js (hunk 4 above)
Comment (asherifhegazy) [reply app/components/products/productsService.js]: it's changed but I don't know why it appears like that for you

Comment (dosama1integrant) [reply app/components/products/productsService.js]: yes i checked it in the branch itself and it's changed 👍

**Review by dosama1integrant — APPROVED** [review]

**Pull request merged**